provider/lxd/credentials.go

```diff
@@ -399,10 +413,6 @@ func (p environProviderCredentials) finalizeRemoteCredential(
 	}
 
 	credAttrs := credentials.Attributes()
-	trustPassword, ok := credAttrs[credAttrTrustPassword]
-	if !ok {
-		return nil, errors.NotValidf("missing %q attribute", credAttrTrustPassword)
-	}
```

Comment:
Do we need something similar to this for both missing trust password or token, or does the check further down suffice?

Reply:
Indeed, the check further suffices. https://github.com/juju/juju/pull/18626/files#diff-09949185d8c8d35f6e2e507f81d38fe45ff48a6e7501430e4be580589c61eb5bR454 is the correct returned error.